zstacklib/zstacklib/utils/file_downloader.py

```
@@ -86,22 +88,32 @@ def _exit(self, exc_type, exc_val, exc_tb):
     def download(self):
-        src_path = self.downloader.cmd.url.lstrip('file:')
+        url = self.downloader.cmd.url
+        if url.startswith('file://'):
+            src_path = url[len('file://'):]
+        elif url.startswith('file:'):
+            src_path = url[len('file:'):]
+        else:
+            src_path = url
         src_path = os.path.normpath(src_path)
+        if not os.path.isabs(src_path):
+            raise Exception('file url must use absolute path: %s' % src_path)
```

Ruff (0.15.4)

[warning] Create your own exception

(TRY002)